src/sinks/util/unix.rs

```diff
@@ -41,27 +40,23 @@ impl UnixSinkConfig {
 
         let sink =
             Box::new(sink.with_flat_map(move |event| iter_ok(encode_event(event, &encoding))));
-        let healthcheck = unix_healthcheck(self.path.clone());
+        let healthcheck = healthcheck(self.path.clone()).boxed().compat();
 
-        Ok((sink, healthcheck))
+        Ok((sink, Box::new(healthcheck)))
     }
 }
 
 #[derive(Debug, Snafu)]
 enum HealthcheckError {
     #[snafu(display("Connect error: {}", source))]
-    ConnectError { source: std::io::Error },
+    ConnectError { source: tokio::io::Error },
 }
 
-fn unix_healthcheck(path: PathBuf) -> Healthcheck {
-    // Lazy to avoid immediately connecting
-    let check = future::lazy(move || {
```

Comment:
I think the reasoning for `lazy` is still valid here unless you've verified that `connect` doesn't bind to the port until polled.

Reply:
Good note. I actually do not understand why new futures need `lazy` function, because new futures not executed while not pulled, for example: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=1903bb7388f8d2965ee127bf8db2675c

Reply:
I think it's because Rust can't yet pass bare async closures around, thus this `future::lazy`.

Here: https://docs.rs/futures-util/0.3.5/src/futures_util/future/lazy.rs.html#34-38, it's very similar to `poll!` macro: https://docs.rs/futures-util/0.3.5/src/futures_util/async_await/poll.rs.html#13-17